django/db/models/sql/query.py

```diff
@@ -2269,8 +2269,21 @@ def add_fields(self, field_names, allow_m2m=True):
                     join_info.joins,
                     join_info.path,
                 )
-                for target in targets:
-                    cols.append(join_info.transform_function(target, final_alias))
+                if (
+                    getattr(join_info.final_field, "is_relation", False)
+                    and len(join_info.targets) > 1
+                ):
+                    cols.append(
+                        ColPairs(
+                            final_alias,
```

Comment:
If we're not calling `join_info.transform_function` we should make sure to consider `self.alias_cols`.

The fact we're not calling it also makes me think we might be missing something regarding transforms. I guess that's not an issue now as related fields don't support transforms (e.g. `user__` translates to a JOIN).